clab/config.go

```diff
@@ -433,6 +436,35 @@ func (c *CLab) verifyLinks() error {
 	return nil
 }
 
+// verifyDuplicateAddresses checks that every static IP address in the topology is unique
+func (c *CLab) verifyDuplicateAddresses() error {
+	var ipv4_list []string
+	var ipv6_list []string
+	for _, node := range c.Nodes {
+		ipv4_addr := node.Config().MgmtIPv4Address
+		if ipv4_addr != "" {
+			for i := range ipv4_list {
+				if ipv4_list[i] == ipv4_addr {
+					return fmt.Errorf("management IPv4 address %q appeared more than once in the topology file", ipv4_addr)
+				}
+			}
+			ipv4_list = append(ipv4_list, ipv4_addr)
```

Comment:
this is a working solution, but not the optimal one
to check for duplicates you rather need to use a map as this will have constant search times, as opposed to the ranging through a list

see example https://replit.com/@RomanDodin/dupsDetection#main.go